packages/next/src/server/async-storage/work-store.ts

```diff
@@ -97,6 +98,7 @@ export function createWorkStore({
    * coalescing, and ISR continue working as intended.
    */
   const isStaticGeneration =
+    !renderOpts.shouldWaitOnAllReady &&
```

Comment:
work store needs to be synced with the conditions from base-server.  When isSSG is turned off in base-server, `workStore.isStaticGeneration` should be aligned